primer3/thermoanalysis.pyx

```diff
@@ -219,41 +224,118 @@ cdef class ThermoResult:
         }
 
 
+def _try_get_dict_int(
+        argname: str,
+        key: Union[str, int],
+        dict_obj: Dict[str, int],
+) -> int:
+    '''Helper function to translate an string name to an integer arguments
+    using a dictionary:
+
+    Args:
+        argname:
+        key: integer value or string name key mapping to an integer
+        dict_obj: dictionary mapping the string to an int
+
+    Returns:
+        integer value for the key in the map
+
+    Raises:
+        KeyError: Key missing in the map
+        TypeError: invalid type for the key
+    '''
+    if isinstance(key, (int, long)):
+        return key
+    elif isinstance(key, str):
+        if key not in dict_obj:
+            raise ValueError(f'{argname}: {key} argument not in {dict_obj}')
+        return dict_obj[key]
+    else:
+        raise TypeError(f'{argname}: {key} invalid type {type(key)}')
+
+
 cdef class ThermoAnalysis:
     ''' Python class that serves as the entry point for thermodynamic
     calculations. Should be instantiated with the proper thermodynamic
     parameters for seqsequence calculations (salt concentrations, correction
     methods, limits, etc.). See module docstring for more information.
     '''
 
-    TM_METHODS = {
+    tm_methods_dict = {
```

Comment:
What would you think about making these actual `enum`s now that we are Python 3.8+?

Reply:
`enum`s  are ok but they end up being more code than just using `dict`s  and aren't faster I think since you end up doing a dictionary look up anyway, but you need to cast the value to an integer as an extra step.

Reply:
like `self.tm_methods_enum['santalucia'].value`